main.go

```diff
@@ -118,24 +128,33 @@ func main() {
 			Aliases: []string{"sn"},
 			Usage:   "Show the clique snapshot",
 			Action: func(c *cli.Context) {
-				GetSnapshot(network, rpcUrl)
+				GetSnapshot(rpcUrl)
 			},
 		},
 	}
 	app.Run(os.Args)
 }
 
-func getRPCURL(network, rpcURL string) string {
-
+// getRPCURL resolves the rpcUrl from the user specified options, or quits if an illegal combination or value is found.
+func getRPCURL(network, rpcURL string, testnet bool) string {
 	if rpcURL != "" {
 		if network != "" {
 			log.Fatalf("Cannot set both rpcURL %q and network %q", rpcURL, network)
 		}
+		if testnet {
+			log.Fatalf("Cannot set both rpcURL %q and testnet", rpcURL)
+		}
 	} else {
+		if testnet {
+			if network != "" {
+				log.Fatalf("Cannot set both network %q and testnet", network)
+			}
+			network = "testnet"
+		}
 		switch network {
 		case "testnet":
 			rpcURL = "https://testnet-rpc.gochain.io"
-		case "mainnet":
+		case "mainnet","":
```

Comment:
This makes `mainnet` the default when network is empty (as well as rpc and testnet).